types/types.go

```diff
@@ -273,6 +273,14 @@ func (e *NotFoundError) Error() string {
 	return fmt.Sprintf("cannot find %v", e.Name)
 }
 
+type ErrorNotNow struct {
+	Reason string
+}
+
+func (e *ErrorNotNow) Error() string {
+	return fmt.Sprintf("Current state prevents this: %v", e.Reason)
```

Comment:
nit:
1. `Error` might be a bit broad as looks like it also wraps state errors around the reason. Maybe we can use a more specific method name like `ErrorInvalidState`. 
1. Consider using lowercase text for better readability.
```
[longhorn-manager-rjzn8] E1017 21:53:38.219480       1 setting_controller.go:205] failed to sync setting for longhorn-system/v2-data-engine: current state prevents this: cannot apply v2-data-engine setting to Longhorn workloads when there are attached volumes
```

Reply:
Is this correct that this method is a dummy method exists only to make the ErrorNotNow an interface?

Reply:
This method exists to satisfy the error interface. The only requirement for Go to treat a type as if it in an error (i.e. allow a function to return that type when an error return is called for) is for that type to implement the Error() method.

It is not a dummy method. Any time string formatting is used to print the error object, Error() will be invoked.

https://pkg.go.dev/fmt

> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

Reply:
Right, this is just to make an error type that can be distinguished from other errors.  I would rather not call it ErrorInvalidState, because that's a bit ambiguous.  It's not the state that is invalid, it is the action we are thinking about taking in the current state.  I could go for ErrorNotInCurrentState, for instance.

Reply:
Thanks James and Eric

Reply:
Renamed error type to ErrorNotWithAttached